### PR TITLE
Folder and file

### PR DESCRIPTION
**Pull request by heitara**
# Unify the File System

Create a proper simulation of a FS in the local storage

### CODE REVIEW

File: src/actions/acts.js [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+const combined = {
+  ...startmenuDefState,
+  ...sidepaneDefState,
+  ...taskbarDefState,
+  ...wallpaperDefState,
+  ...menuDefState,
+  ...deskDefState,
+  ...widpaneDefState,
+};
+
```
Comment (heitara) [inline src/actions/acts.js:22]: If you check the [Chain Of Responsibility Design pattern](https://refactoring.guru/design-patterns/chain-of-responsibility/typescript/example), you will see how we can split this massive reducer in several small ones.

File: src/actions/actsDefState.js [hunk 2]
```diff
@@ -0,0 +1,381 @@
...
+];
+
+const themes = ["default", "dark", "ThemeA", "ThemeB", "ThemeD", "ThemeC"];
+
+export const wallpaperDefState = {
+  themes: themes,
+  wps: wps,
+  src: walls[wps],
+  locked: !(locked == "false"),
+  booted: false || import.meta.env.MODE == "development",
```
Comment (heitara) [inline src/actions/actsDefState.js]:
```suggestion
  booted: import.meta.env.MODE == "development",
```

File: src/reducers/files.js [hunk 3]
```diff
@@ -16,29 +16,38 @@ const fileReducer = (state = defState, action) => {
...
-    tmp.hid++;
-    if (tmp.hid > tmp.hist.length - 1) tmp.hid = tmp.hist.length - 1;
-    navHist = true;
+  switch (action.type) {
+    case "FILEDIR":
+      tmp.cdir = action.payload;
+      break;
+    case "FILEPATH":
+      var pathid = tmp.data.parsePath(action.payload);
+      if (pathid) tmp.cdir = pathid;
```
Comment (heitara) [inline src/reducers/files.js]:
```suggestion
      if (pathid) {
          tmp.cdir = pathid;
      }
```

File: src/reducers/files.js [hunk 4]
```diff
@@ -16,29 +16,38 @@ const fileReducer = (state = defState, action) => {
...
+      tmp.view = action.payload;
+      break;
+    case "FILEPREV":
+      tmp.hid--;
+      if (tmp.hid < 0) tmp.hid = 0;
+      navHist = true;
+      break;
+    case "FILENEXT":
+      tmp.hid++;
+      if (tmp.hid > tmp.hist.length - 1) tmp.hid = tmp.hist.length - 1;
```
Comment (heitara) [inline src/reducers/files.js]:
```suggestion
      if (tmp.hid > tmp.hist.length - 1) {
          tmp.hid = tmp.hist.length - 1;
      }
```

Comment (NakataCode) [reply src/reducers/files.js]: Fixed

File: src/reducers/files.js (hunk 3 above)
Comment (NakataCode) [reply src/reducers/files.js]: Fixed

File: src/actions/actsDefState.js (hunk 2 above)
Comment (NakataCode) [reply src/actions/actsDefState.js]: Fixed

**Pull request merged**